### PR TITLE
Deprecate the `license.enterprise` setting

### PR DESCRIPTION
**Pull request by mfussenegger**
## Summary of the changes / Why this improves CrateDB

See commit messages.

Note that not everything works correctly yet using the community edition
(E.g. since the `license.enterprise` flag is still there and defaults to
true, the license service will generate a trial license and the
expiration logic is in place). We'll have to follow up on this to take
the availability of the enterprise modules into consideration.

## Checklist

 - [x] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

Co-authored-by: Savvas Makalias <savvas@crate.io>

### CODE REVIEW

**Comment by mfussenegger** [comment]
@lukas-bernert Can you take a look at the documentation part of this PR?

File: blackbox/docs/editions/enterprise.rst [hunk 1]
```diff
@@ -33,6 +32,12 @@ enterprise features are enabled:
...
 
+
+.. note::
+
+   It is also possible to build a :ref:`community-edition` which won't contain
+   these features but which can be used without licensing restrictions.
+
 .. _enterprise_trial:
 
 Trial
```
Comment (mfussenegger) [inline blackbox/docs/editions/enterprise.rst]: Note that this isn't updated yet because the logic change hasn't been done/merged yet

Comment (smakalias) [reply blackbox/docs/editions/enterprise.rst]: makes sense now that you mentioned - I was about to ask :)

File: blackbox/docs/config/node.rst [hunk 2]
```diff
@@ -680,6 +680,11 @@ Enterprise License
...
 
-  Setting this to ``false`` disables the `Enterprise Edition`_ of CrateDB.
+  This is a deprecated setting that allows you to disable the `Enterprise
+  Edition`_ features of CrateDB, enabling you to use all community edition
+  features of CrateDB without restrictions or license.
+  The setting will be removed in CrateDB 4.0 and instead you'll have to
+  `Acquire a license`_ or switch to the :ref:`community-edition` distribution.
 
 .. _`Enterprise Edition`: https://crate.io/enterprise-edition/
+.. _`Acquire a license`: https://crate.io/license-update/
```
Comment (lukas-bernert) [inline blackbox/docs/config/node.rst]: I think it is better to link to this site:
https://crate.io/pricing/

The license-update page is mainly for existing customers with an expired license.

File: blackbox/docs/editions/enterprise.rst (hunk 1 above)
Comment (lukas-bernert) [inline blackbox/docs/editions/enterprise.rst]: Shouldn't we depreacate the Trial as well as it will be removed with 4.0?

Comment (mfussenegger) [reply blackbox/docs/editions/enterprise.rst]: See my comment here https://github.com/crate/crate/pull/8345#discussion_r263794126  - we'll update this once we've the logic change in place.

**Pull request merged**